ibm/service/backuprecovery/data_source_ibm_backup_recovery_protection_policies.go

```diff
@@ -0,0 +1,5455 @@
+// Copyright IBM Corp. 2024 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+/*
+ * IBM OpenAPI Terraform Generator Version: 3.94.0-fa797aec-20240814-142622
+ */
+
+package backuprecovery
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/ibm-backup-recovery-sdk-go/backuprecoveryv1"
+)
+
+func DataSourceIbmBackupRecoveryProtectionPolicies() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIbmBackupRecoveryProtectionPoliciesRead,
+
+		Schema: map[string]*schema.Schema{
+			"x_ibm_tenant_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Specifies the key to be used to encrypt the source credential. If includeSourceCredentials is set to true this key must be specified.",
+			},
+			"request_initiator_type": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Specifies the type of request from UI, which is used for services like magneto to determine the priority of requests.",
+			},
+			"ids": &schema.Schema{
```

Comment:
May be generic comment check for all datasources/resources

Reply:
Yes @hkantare 
We can provide both ids and policy_names together. The API first filter out the policies based on ids and then proceed to match the names defined in policy_names. If a policy matching both the values is present then it is listed otherwise we get an empty policy list. In this case the API does not throw any error.